storage/innobase/dict/dict0load.cc

```diff
@@ -3224,3 +3224,55 @@ dict_load_foreigns(
 	ut_free(pcur.old_rec_buf);
 	DBUG_RETURN(DB_SUCCESS);
 }
+
+/** Get the system tablespace index root page
+@param root_pages list of index root pages
+@return error code */
+dberr_t dict_get_sys_tablespace_indexes(
+  std::vector<uint32_t> *root_pages)
+{
+  /* Add system table index root pages */
+  dict_sys.sys_tables->add_index_root_pages(root_pages);
+  dict_sys.sys_indexes->add_index_root_pages(root_pages);
+  dict_sys.sys_columns->add_index_root_pages(root_pages);
+  dict_sys.sys_fields->add_index_root_pages(root_pages);
+  dict_sys.sys_foreign->add_index_root_pages(root_pages);
+  dict_sys.sys_foreign_cols->add_index_root_pages(root_pages);
+  dict_sys.sys_virtual->add_index_root_pages(root_pages);
+
+  mtr_t mtr;
+  btr_pcur_t pcur;
+  ulint len;
+  dberr_t err= DB_SUCCESS;
+  mtr.start();
+  dict_sys.lock(SRW_LOCK_CALL);
+  for (const rec_t *rec= dict_startscan_system(
+         &pcur, &mtr, dict_sys.sys_indexes);
+       rec; rec= dict_getnext_system_low(&pcur, &mtr))
+  {
+    const byte *field= rec_get_nth_field_old(
+      rec, DICT_FLD__SYS_INDEXES__SPACE, &len);
+    if (len != 4)
+    {
+      err= DB_CORRUPTION;
+      break;
+    }
+    uint32_t space= mach_read_from_4(field);
+    if (space > 0) continue;
+
+    field= rec_get_nth_field_old(
+      rec, DICT_FLD__SYS_INDEXES__PAGE_NO, &len);
+    if (len != 4)
+    {
+      err= DB_CORRUPTION;
+      break;
+    }
+    uint32_t root_page_no= mach_read_from_4(field);
+    if (root_page_no != FIL_NULL)
+      root_pages->push_back(root_page_no);
```

Comment:
What if there are duplicate `SYS_INDEXES.PAGE` entries? Should `root_pages` rather be a `std::unordered_set`?